### PR TITLE
feat: support ctx in `Cache`

### PR DESCRIPTION
**Pull request by chenquan**
Signed-off-by: chenquan <chenquan.dev@foxmail.com>

### CODE REVIEW

File: core/stores/cache/cache_test.go [hunk 1]
```diff
@@ -45,7 +49,7 @@ func (mc *mockedNode) Get(key string, v interface{}) error {
 }
 
 func (mc *mockedNode) IsNotFound(err error) bool {
-	return err == mc.errNotFound
+	return errors.Is(err, mc.errNotFound)
```
Comment (chenquan) [inline core/stores/cache/cache_test.go:52]: Changed semantics, this modification requires discussion.

File: core/stores/cache/cachenode.go [hunk 2]
```diff
@@ -94,22 +107,32 @@ func (c cacheNode) Get(key string, v interface{}) error {
 
 // IsNotFound checks if the given error is the defined errNotFound.
 func (c cacheNode) IsNotFound(err error) bool {
-	return err == c.errNotFound
+	return errors.Is(err, c.errNotFound)
```
Comment (chenquan) [inline core/stores/cache/cachenode.go:110]: Changed semantics, this modification requires discussion.

File: core/stores/cache/cache.go [hunk 3]
```diff
@@ -85,52 +116,87 @@ func (cc cacheCluster) Del(keys ...string) error {
...
 	}
 
-	return c.(Cache).Get(key, v)
+	return c.(Cache).GetCtx(ctx, key, v)
 }
 
+// IsNotFound checks if the given error is the defined errNotFound.
 func (cc cacheCluster) IsNotFound(err error) bool {
-	return err == cc.errNotFound
+	return errors.Is(err, cc.errNotFound)
```
[the remark this answers is not in the document]
Comment (kevwan) [reply core/stores/cache/cache.go:136]: It's ok.

File: core/stores/cache/cache_test.go (hunk 1 above)
Comment (kevwan) [reply core/stores/cache/cache_test.go:52]: no problem.

**Pull request merged**